ledger-core/insolar/payload/validation_vdelegatedrequestfinished.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/assured-ledger/blob/master/LICENSE.md.
+
+package payload
+
+import (
+	"github.com/insolar/assured-ledger/ledger-core/insolar/contract"
+	"github.com/insolar/assured-ledger/ledger-core/pulse"
+	"github.com/insolar/assured-ledger/ledger-core/vanilla/throw"
+)
+
+var _ Validatable = &VDelegatedRequestFinished{}
+
+func (m *VDelegatedRequestFinished) validateUnimplemented() error {
+	switch {
+	case m.ResultFlags != nil:
+		return throw.New("ResultFlags should be empty")
+	case m.EntryHeadHash != nil:
+		return throw.New("EntryHeadHash should be empty")
+	}
+	return nil
+}
+
+func (m *VDelegatedRequestFinished) isIntolerable() bool {
+	return m.GetCallFlags().GetInterference() == contract.CallIntolerable
+}
+
+func (m *VDelegatedRequestFinished) Validate(currentPulse pulse.Number) error {
+	if err := m.validateUnimplemented(); err != nil {
+		return err
+	}
+
+	switch m.GetCallType() {
+	case CTMethod, CTConstructor:
+	case CTInboundAPICall, CTOutboundAPICall, CTNotifyCall, CTSAGACall, CTParallelCall, CTScheduleCall:
+		return throw.New("CallType is not implemented")
+	case CTInvalid:
+		fallthrough
+	default:
+		return throw.New("CallType must be valid")
+	}
+
+	callFlags := m.GetCallFlags()
+	if f := callFlags.GetState(); f >= contract.StateFlagCount || f <= contract.StateInvalid {
```

Comment:
I'd move this validation to method on stateFlag, so there would be callFlags.GetState().IsValid()

And I'm not sure why these checks are in one if else if

Reply:
Sure, let's do it that way. It's better.
They're grouped since it's one field.

Reply:
hm, maybe then do a method on CallFlags?

Reply:
I thought of three methods IsValid on State, IsValid on Interference and IsValid on CallFlags that'll group both.

Reply:
fixed